### PR TITLE
web-wallet: Add BEP20 address and memo validation

### PR DESCRIPTION
**Pull request by nortonandreev**
Resolves dusk-network/web-wallet#508

This PR adds validation for BEP20 bridge operations, ensuring users provide valid EVM addresses when sending to the bridge address.

##  **Key Features**

- **Bridge Detection**: Automatically detects when sending to BEP20 bridge address
- **Required Memo**: Makes memo field mandatory for bridge operations with EVM address validation
- **EVM Address Validation**: New `isValidEvmAddress()` utility validates 40-character hex addresses (with/without "0x")
- **Enhanced UX**: Clear error messages, visual feedback, and contextual banners guide users

## **Main Changes**

- `Send.svelte` - Added bridge detection logic and memo validation
- `isValidEvmAddress.js` - New EVM address validation utility with comprehensive tests
- Enhanced error handling and user feedback for bridge operations

## **Validation Rules**

- **Bridge Operations**: Memo required and must be valid EVM address
- **Regular Sends**: Memo remains optional
- **EVM Format**: 40 hex characters, optionally prefixed with "0x"

## **Testing**

- 92 test cases covering EVM address validation edge cases
- Updated component snapshots for UI changes

### CODE REVIEW

File: web-wallet/src/routes/(app)/dashboard/send/+page.svelte [hunk 1]
```diff
@@ -45,6 +45,8 @@
     }
   }
 
+  const BEP20_BRIDGE_ADDRESS = import.meta.env.VITE_BEP20_BRIDGE_ADDRESS;
```
[the remark this answers is not in the document]
Comment (HDauven) [reply web-wallet/src/routes/(app)/dashboard/send/+page.svelte:48]: @nortonandreev I think this approach is fine. Theoretically we can set it to the same address on all networks, because It's just an address that is being observed and reacted to depending on certain conditions. 

On testnet it's been tested against arbitrary addresses.

**Review by kieranhall — APPROVED** [review]
